man/en/sosreport.1

```diff
@@ -247,6 +249,60 @@ output, or string data from the system. The resulting logs may be used
 to understand the actions that sos would have taken without the dry run
 option.
 .TP
+.B \--upload
+If specified, attempt to upload the resulting archive to a vendor defined location.
+
+This option is implied if --upload-url is used.
+
+You may be prompted for a username and password if these are not defined by the vendor
+as well. If these credentials are not provided, sos will still run and create an archive
+but will not attempt an automatic upload, instead relying on the end user to upload it
+as needed.
+
+The sosreport archive will still remain on the local filesystem even after a successful
+upload.
+
+Note that depending on the distribution sos is being run on, or the vendor policy detected during
+execution, there may be dependencies that are not strictly required by the package
+at installation time.
+
+For example, for HTTPS uploads the python-requests library must be available. If this
+library is not available, HTTPS uploads will not be attempted.
+.TP
+.B \--upload-url URL
+If a vendor does not provide a default upload location, or if you would like to upload
+the archive to a different location, specify the address here.
+
+A support protocol MUST be specified in this URL. Currently uploading is supported
+for HTTPS, SFTP, and FTP protocols.
+
+If your destination server listens on a non-standard port, specify the listening
+port in the URL.
```

Comment:
Not sure if this needs to mention (and if so, if it should be here): Support of HTTP/FTP proxy fully relies on HTTP[|S]_PROXY environment variables only. Sosreport itself does not allow configuring a proxy server to upload the report via it.

Leaving on Jake if this not is worth to mention or not. I am fully OK in not explicitly mentioning it anywhere.